internal/super/generate.go

```diff
@@ -127,17 +137,12 @@ func generateNew(
 	}
 
 	name := args[0]
-	var filename string
-
-	// Don't add .cdc extension if it's already there
-	if strings.HasSuffix(name, ".cdc") {
-		filename = name
-	} else {
-		filename = fmt.Sprintf("%s.cdc", name)
-	}
+	var filename = addCDCExtension(name)
```

Comment:
nit: It seems like to me that the generate command is meant to accept both names with & without exentions.  (i.e. extension is only added if not already there)

However, the variable `name` is used throughout this function for templating, flow.json filling, etc.  Do we need to strip `.cdc` from `name` as well as add it to `filename`?

i.e. When I run `flow generate contract Foobar.cdc` what happens?  If this is intended to be valid syntax, can we add a test case?

Reply:
Fixed good catch